Upstream: yes, in 2.17 branch and master
From fc1abbe2ccbcc8b8cb4525e3838e459ff3841635 Mon Sep 17 00:00:00 2001
From: Andreas Jaeger <aj@suse.de>
Date: Fri, 11 Jan 2013 11:53:13 +0100
Subject: [PATCH] Add MSG_FASTOPEN

	[BZ #15003]
	* sysdeps/unix/sysv/linux/bits/socket.h (MSG_FASTOPEN): New
	value. Sync with Linux 3.7.
(cherry picked from commit c6fe55cf6089fc5cf1cea15fc7e1c9a8b90d9fda)
---
 sysdeps/unix/sysv/linux/bits/socket.h |    5 +++--
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/bits/socket.h b/sysdeps/unix/sysv/linux/bits/socket.h
index df8f167..eadd7d9 100644
--- a/sysdeps/unix/sysv/linux/bits/socket.h
+++ b/sysdeps/unix/sysv/linux/bits/socket.h
@@ -1,6 +1,5 @@
 /* System-specific socket constants and types.  Linux version.
-   Copyright (C) 1991, 1992, 1994-2001, 2004, 2006-2010, 2011, 2012
-   Free Software Foundation, Inc.
+   Copyright (C) 1991-2013 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -208,6 +207,8 @@ enum
 #define	MSG_MORE	MSG_MORE
     MSG_WAITFORONE	= 0x10000, /* Wait for at least one packet to return.*/
 #define MSG_WAITFORONE	MSG_WAITFORONE
+    MSG_FASTOPEN	= 0x20000000, /* Send data in TCP SYN.  */
+#define MSG_FASTOPEN	MSG_FASTOPEN
 
     MSG_CMSG_CLOEXEC	= 0x40000000	/* Set close_on_exit for file
 					   descriptor received through
-- 
1.7.3.4

